src/compiler/scala/tools/nsc/classpath/AggregateClassPath.scala

```diff
@@ -137,12 +139,14 @@ case class AggregateClassPath(aggregates: Seq[ClassPath]) extends ClassPath {
       val name = entry.name
       if (indices.containsKey(name)) {
         val index = indices.get(name)
-        val existing = mergedEntries(index)
+        var merges = mergedEntries(index)
+
+        if (entry.binary.isDefined && (merges.binary.isEmpty || entry.binary.get.hasTastyExtension))
```

Comment:
probably irrelevant, but keep the first tasty:

```suggestion
        if (
          entry.binary.exists(_.hasTastyExtension) && !merges.binary.exists(_.hasTastyExtension) ||
          entry.binary.isDefined && merges.binary.isEmpty)
```

Reply:
removed